internal/client/blueprints_test.go

```diff
@@ -1229,3 +1229,48 @@ func TestBlueprintChangeV1(t *testing.T) {
 	require.NotNil(t, bp, "GET blueprint change failed: missing blueprint")
 	assert.Equal(t, bp.Version, "0.0.1")
 }
+
+func TestEmptyPackageNameBlueprintJsonV0(t *testing.T) {
+	for _, packageSnippet := range []string{
+		`"packages": [{"name": "", "version": "*"}]`,
+		`"packages": [{"name": ""}]`,
+	} {
+		bp := `{
+		"name": "test-emptypackage-blueprint-v0",
+		"description": "TestEmptyPackageNameBlueprintV0",
+		"version": "0.0.1",
+        ` + packageSnippet + `}`
+
+		expectedErrorPrefix := "BlueprintsError: All package entries need to contain the name of the package."
```

Comment:
This is nice, thank you! It seems with a tiny bit of extra effort we can test the exact error here too, I had a quick look and something like this might work? This would give us this nice extra confidence that both cases are handled correctly and in case we refactor that we still are confident about t it :)
```go
index b96dc466e..997a048f7 100644
--- a/internal/client/blueprints_test.go
+++ b/internal/client/blueprints_test.go
@@ -1231,25 +1231,32 @@ func TestBlueprintChangeV1(t *testing.T) {
 }
 
 func TestEmptyPackageNameBlueprintJsonV0(t *testing.T) {
-	for _, packageSnippet := range []string{
-		`"packages": [{"name": "", "version": "*"}]`,
-		`"packages": [{"name": ""}]`,
+	for _, tc := range []struct {
+		packageSnippet string
+		specificErr    string
+	}{
+		{
+			`"packages": [{"name": "", "version": "*"}]`,
+			"Entry #1 has version '*' but no name.",
+		}, {
+			`"packages": [{"name": ""}]`,
+			"Entry #1 has no name.",
+		},
 	} {
 		bp := `{
 		"name": "test-emptypackage-blueprint-v0",
 		"description": "TestEmptyPackageNameBlueprintV0",
 		"version": "0.0.1",
-        ` + packageSnippet + `}`
+        ` + tc.packageSnippet + `}`
 
 		expectedErrorPrefix := "BlueprintsError: All package entries nee
d to contain the name of the package."
+		expectedErr := expectedErrorPrefix + " " + tc.specificErr
 
 		resp, err := PostJSONBlueprintV0(testState.socket, bp)
 		require.NoError(t, err, "failed with a client error")
 		require.False(t, resp.Status, "Negative status expected.")
 		require.Equal(t, len(resp.Errors), 1, "There should be exactly o
ne error")
-		reasoning := "I expected an error message starting with\n" + exp
ectedErrorPrefix +
-			"\nnot\n" + resp.Errors[0].String()
-		assert.True(t, strings.HasPrefix(resp.Errors[0].String(), expect
edErrorPrefix), reasoning)
+		assert.Equal(t, resp.Errors[0].String(), expectedErr)
 	}
 }
 
```